### PR TITLE
feat: Introduce `native_type_declaration_casing` fixer

### PR DESCRIPTION
**Pull request by SpacePossum**
replaces https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7130

- introduce `native_type_declaration_casing`
- add support for class constants (PHP8.3)
- add support for class properties
- deprecate `native_function_type_declaration_casing`

--edit

not sure how to fix the tests...

### CODE REVIEW

**Comment by Wirone** [comment]
> not sure how to fix the tests...

You have `VersionSpecification(8_03_00)` but PHP8.3 is not supported yet. You need to add it [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/f1abe952e08402721fc20bea2ab3892dfbb97ef2/tests/Test/AbstractFixerTestCase.php#L159).

File: doc/list.rst [hunk 1]
```diff
@@ -1324,9 +1324,16 @@ List of Available Rules
 
    Native type hints for functions should use the correct case.
 
-   Part of rule sets `@PhpCsFixer <./ruleSets/PhpCsFixer.rst>`_ `@Symfony <./ruleSets/Symfony.rst>`_
+   *warning deprecated*   Use ``native_type_declaration_casing`` instead.
 
    `Source PhpCsFixer\\Fixer\\Casing\\NativeFunctionTypeDeclarationCasingFixer <./../src/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixer.php>`_
+-  `native_type_declaration_casing <./rules/casing/native_type_declaration_casing.rst>`_
+
+   Native type hints should be used in the correct case.
```
Comment (localheinz) [inline doc/list.rst:1332]:
```suggestion
   Native type declarations should be used in the correct case.
```

Comment (SpacePossum) [reply doc/list.rst:1332]: thanks, will update, I'll update the other one as well as we are touching the same files ; )

Comment (localheinz) [reply doc/list.rst:1332]: Thank you, @SpacePossum - I have opened a pull request updating additional mentions in #7338!

**Comment by Wirone** [comment]
> maybe it would be better to derive the supported versions used by the tests from `.github/workflows/ci.yml`?

@SpacePossum It was [discussed before](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7212#discussion_r1297361235) (well, kind of), but even if we want to change the source of supported PHP version, it should be done in the separate PR, so for now we need to extend range of supported versions where the source is.

File: doc/list.rst (hunk 1 above)
Comment (SpacePossum) [reply doc/list.rst]: Thanks so much :+1:  :)

**Comment by SpacePossum** [comment]
force-pushed to rebase on lastest main, no other changes, after that added two commits;
one to following the suggestion to update the docs with better wording (thanks @localheinz) , 
second to update the tests to allow PHP8.3 (thanks @Wirone)
lets see what the what the CI thinks :crossed_fingers:

**Review by keradus — APPROVED** [review]
nice!

**Comment by SpacePossum** [comment]
nice, thanks!

**Pull request merged**